File: android/src/main/kotlin/com/jarvan/fluwx/handler/FluwxSubscribeMsgHandler.kt
package com.jarvan.fluwx.handler

import com.tencent.mm.opensdk.modelbiz.SubscribeMessage
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel



/// create 2018/12/20 by cai


class FluwxSubscribeMsgHandler {
    fun subScribeMsg(call: MethodCall, result: MethodChannel.Result) {
        val appId = call.argument<String>("appId")
        val scene = call.argument<Int>("scene")
        val templateId = call.argument<String>("templateId")
        val reserved = call.argument<String>("reserved")

        val req = SubscribeMessage.Req()
        req.openId = appId
        req.scene = scene!!
        req.reserved = reserved
        req.templateID = templateId
        val b = WXAPiHandler.wxApi?.sendReq(req)
        result.success(b)
    }
}

File: ios/Classes/handler/FluwxSubscribeMsgHandler.m
//
//  FluwxSubscribeMsgHandler.m
//  fluwx
//
//  Created by cjl on 2018/12/19.
//

#import "FluwxSubscribeMsgHandler.h"
#import <WXApiRequestHandler.h>

@implementation FluwxSubscribeMsgHandler {
    NSObject <FlutterPluginRegistrar> *_fluwxRegistrar;
}
- (instancetype)initWithRegistrar:(NSObject <FlutterPluginRegistrar> *)registrar {
    self = [super init];
    if (self) {
        _fluwxRegistrar = registrar;
    }
    return self;
}

- (void)handleSubscribeWithCall:(FlutterMethodCall *)call result:(FlutterResult)result {
    NSDictionary *params = call.arguments;
    NSString *appId = [params valueForKey:@"appId"];
    int scene = [[params valueForKey:@"scene"] intValue];
    NSString *templateId = [params valueForKey:@"templateId"];
    NSString *reserved = [params valueForKey:@"reserved"];

    WXSubscribeMsgReq *req = [WXSubscribeMsgReq new];
    req.scene = (UInt32) scene;
    req.templateId = templateId;
    req.reserved = reserved;
    req.openID = appId;

    [WXApi sendReq:req completion:^(BOOL done) {result(@(done));}];

   
}


@end
